_source/_posts/2021-10-16-full-stack-java.md

```diff
@@ -0,0 +1,691 @@
+```
+
+Then, in the `return` block, remove the JSX between the `image` property and `album` property and replace it with `{metadataRows}`.
+
+```html
+    <input id="file_image" type="file" onChange={this.onBlobChange(true, 'image')} accept="image/*" />
+  </AvGroup>
+</AvGroup>
+{metadataRows}
+<AvGroup>
+  <Label for="album.title">
+    <Translate contentKey="galleryApp.photo.album">Album</Translate>
+  </Label>
+```
+
+//todo: update for Cypress
+
+In `src/test/javascript/e2e/entities/photo/photo.spec.ts`, remove the code that sets the data in these fields:
+
+```js
+photoUpdatePage.setHeightInput('5');
+expect(await photoUpdatePage.getHeightInput()).to.eq('5');
+photoUpdatePage.setWidthInput('5');
+expect(await photoUpdatePage.getWidthInput()).to.eq('5');
+photoUpdatePage.setTakenInput('01/01/2001' + protractor.Key.TAB + '02:30AM');
+expect(await photoUpdatePage.getTakenInput()).to.contain('2001-01-01T02:30');
+photoUpdatePage.setUploadedInput('01/01/2001' + protractor.Key.TAB + '02:30AM');
+expect(await photoUpdatePage.getUploadedInput()).to.contain('2001-01-01T02:30');
+```
+
+You can also remove all the getters and setters for these fields in `src/test/javascript/e2e/entities/photo/photo-update.page-object.ts`:
+
+```js
+setHeightInput(height) {
+  this.heightInput.sendKeys(height);
+}
+
+getHeightInput() {
+  return this.heightInput.getAttribute('value');
+}
+
+setWidthInput(width) {
+  this.widthInput.sendKeys(width);
+}
+
+getWidthInput() {
+  return this.widthInput.getAttribute('value');
+}
+
+setTakenInput(taken) {
+  this.takenInput.sendKeys(taken);
+}
+
+getTakenInput() {
+  return this.takenInput.getAttribute('value');
+}
+
+setUploadedInput(uploaded) {
+  this.uploadedInput.sendKeys(uploaded);
+}
+
+getUploadedInput() {
+  return this.uploadedInput.getAttribute('value');
+}
+```
+
+Stop your Maven process, run `npm webapp:build`, start Maven again, and run `npm run e2e` to make sure everything still works. 
+
+**NOTE**: If you experience authentication errors in your Cypress tests, it's likely because you've violated Auth0's [Rate Limit Policy](https://auth0.com/docs/support/policies/rate-limit-policy). As a workaround, I recommend you use Keycloak for Cypress tests. You can do this by opening a new terminal window and starting your app there (without running `source .auth0.env`).
+
+If you upload an image you took with your smartphone, the height, width, and taken values should all be populated. If they're not, chances are your image doesn't have the data in it.
+
+> Need some sample photos with EXIF data? You can download pictures of my 1966 VW Bus from [an album on Flickr](https://www.flickr.com/photos/mraible/albums/72157689027458320).
+
+## Add a React Photo Gallery
+
+You've added metadata extraction to your backend, but your photos still display in a list rather than in a grid (like Flickr). To fix that, you can use [React Photo Gallery](https://github.com/neptunian/react-photo-gallery) component. Install it using npm:
```

Comment:
```suggestion
You've added metadata extraction to your backend, but your photos still display in a list rather than in a grid (like Flickr). To fix that, you can use the [React Photo Gallery](https://github.com/neptunian/react-photo-gallery) component. Install it using npm:
```
adding an article